### PR TITLE
8397 - related secured party code dropdown

### PR DESCRIPTION
**Pull request by jordiwes**
*Issue #:* /bcgov/entity#8397

*Description of changes:*
- for other types of registration, only allow related secured party code

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/parties/SecuredParties.vue [hunk 1]
```diff
@@ -298,30 +430,46 @@ export default defineComponent({
...
 }
 .v-remove:hover {
   background-color: white important;
 }
+
+.party-search {
+  background-color: white;
+}
+.auto-complete-row {
+  font-size: 14px;
```
Comment (kialj876) [inline ppr-ui/src/components/parties/SecuredParties.vue]: use rem for font pls

File: ppr-ui/src/components/parties/SecuredParties.vue [hunk 2]
```diff
@@ -298,30 +430,46 @@ export default defineComponent({
...
+}
+
+::v-deep .party-search .v-select__selections {
+  color: $gray7 !important;
+}
+
+::v-deep .v-list-item--active {
+  padding-left: 30px;
+  color: $primary-blue !important;
+  font-size: 14px;
```
Comment (kialj876) [inline ppr-ui/src/components/parties/SecuredParties.vue]: same here

File: ppr-ui/tests/unit/test-data/mock-new-registration.ts [hunk 3]
```diff
@@ -55,6 +55,16 @@ export const mockedMarriageMH: RegistrationTypeIF = {
   text: ''
 }
 
+export const mockedOtherCarbon: RegistrationTypeIF = {
```
Comment (kialj876) [inline ppr-ui/tests/unit/test-data/mock-new-registration.ts]: is there a reason for these mocked registration types other than getting the value we already have defined elsewhere? Could you grab this object from the list of registration types in resources instead? That way we don't have more than one place to change it if we need to alter it later

**Review by kialj876 — COMMENTED** [review]
Looks good, just a couple small things

File: ppr-ui/tests/unit/test-data/mock-new-registration.ts (hunk 3 above)
Comment (jordiwes) [reply ppr-ui/tests/unit/test-data/mock-new-registration.ts]: We're using this in tests to set the store to test the other registration type changes... it needs to be a RegistrationTypeIF to set the store... if that makes sense

**Comment by kialj876** [comment]
hey steph you can probably fix that build error with the npm run lint cmd ?

**Review by doug-lovett — APPROVED** [review]

File: ppr-ui/src/components/parties/SecuredParties.vue (hunk 1 above)
Comment (jordiwes) [reply ppr-ui/src/components/parties/SecuredParties.vue]: done

File: ppr-ui/src/components/parties/SecuredParties.vue (hunk 2 above)
Comment (jordiwes) [reply ppr-ui/src/components/parties/SecuredParties.vue]: done

File: ppr-ui/tests/unit/test-data/mock-new-registration.ts (hunk 3 above)
Comment (jordiwes) [reply ppr-ui/tests/unit/test-data/mock-new-registration.ts]: done

**Review by kialj876 — APPROVED** [review]
thanks for making those changes!

**Pull request merged**